packages/react-reconciler/src/__tests__/ReactSuspenseWithNoopRenderer-test.js

```diff
@@ -4203,4 +4203,60 @@ describe('ReactSuspenseWithNoopRenderer', () => {
     });
     expect(Scheduler).toHaveYielded(['Unmount Child']);
   });
+
+  it('should resume the suspending component when work loop yields immediately after catching wakeable and when wakeable pings before resuming the work loop', async () => {
+    let cachedFoo;
+    let cachedBar;
+    const resolveCbs = new Set();
+    function useFoo() {
+      if (!cachedFoo) {
+        throw new Promise(resolve =>
+          setTimeout(() => resolve((cachedFoo = 'foo')), 50),
+        );
+      }
+      return cachedFoo;
+    }
+    function useBar() {
+      if (!cachedBar) {
+        const p = {
+          then(cb) {
+            resolveCbs.add(cb);
+            // uncomment to make this test pass
+            // if (cachedBar) {
+            //   p.resolve();
+            // }
+          },
+          resolve() {
+            resolveCbs.forEach(cb => cb());
+          },
+        };
+        Scheduler.unstable_forceYield();
```

Comment:
this is also an important bit mentioned in the point 3 [here](this is the important bit mentioned in the point 6 [here](https://github.com/facebook/react/issues/24864#issuecomment-1178716544))